### PR TITLE
Don't merge `context`, `body`, `json`, `form` & fix types

### LINKED ISSUE #897: 'context' keyword is stripped out from request body
#### Describe the bug

 - Node.js version: 12.11.0
 - OS & version: ubuntu 18.04
 - got version: 10.0.0-alpha.2.2

`context` key is stripped from body.

#### Actual behavior
when we include `context` in our request body it is stripped out from outgoing request.
...

#### Code to reproduce

```js
const got = require('got');
(async () => {
  let resp;
  try {

    resp = await got('http://google.com',{
      method : 'POST',
      json   : {
        context : { a: 'test'},
        a       : { b: 'asdasd'}
      }
    });
  } catch (e) {
    console.log(JSON.parse(e.options.body));
  }

  console.log(resp);
})();
```

#### Checklist

- [x] I have read the documentation.
- [x] I have tried my code with the latest version of Node.js and Got.

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes #897

### CODE REVIEW

File: test/arguments.ts [hunk 1]
```diff
@@ -185,7 +185,7 @@ test('throws TypeError when known `options.hooks` value is not an array', async
 		got('https://example.com', {hooks: {beforeRequest: {}}}),
 		{
 			instanceOf: TypeError,
-			message: 'options.hooks.beforeRequest is not iterable'
+			message: 'Parameter `beforeRequest` must be an array, not Array'
 		}
```
Comment (sindresorhus) [inline test/arguments.ts]: This error message makes no sense. It should be more helpful.

File: test/post.ts [hunk 2]
```diff
@@ -169,7 +169,7 @@ test('throws when form body is not a plain object or array', async t => {
 });
 
 // See https://github.com/sindresorhus/got/issues/897
-test.failing('the `json` payload is not touched', withServer, async (t, server, got) => {
```
Comment (sindresorhus) [inline test/post.ts]: Can you update the test to also check context, form, body, and hooks? Just to ensure we never regress the other properties either.

Comment (szmarczak) [reply test/post.ts]: Done. The tests for `context` already exist in the `arguments.ts` file.

**Comment by sindresorhus** [comment]
Great 🙌

**Pull request merged**